Fix for Support ARGB for pen #393

griffpatch opened the pull request:
Requires fix to scratch-render before this will work.
See:
6f5acfee7b961d71237f1fd50bb3d5a5139c527e

griffpatch committed:
Fix for Support ARGB for pen #393

Requires fix to scratch-render before this will work (see:
6f5acfee7b961d71237f1fd50bb3d5a5139c527e)

thisandagain commented on src/blocks/scratch3_pen.js:
```diff
@@ -258,6 +258,9 @@ Scratch3PenBlocks.prototype.setPenColorToColor = function (args, util) {
     penState.penAttributes.color4f[1] = rgb.g / 255.0;
     penState.penAttributes.color4f[2] = rgb.b / 255.0;
+    if (rgb.hasOwnProperty('a')) {  // Will there always be an 'a'?
+        penState.penAttributes.color4f[3] = rgb.a/255.0;
```
Looks like you have a lint / code style issue here:
```js
penState.penAttributes.color4f[3] = rgb.a / 255.0;
```

thisandagain commented on src/util/color.js:
```diff
@@ -40,10 +40,11 @@ Color.decimalToHex = function (decimal) {
     var g = (decimal >> 8) & 0xFF;
     var b = decimal & 0xFF;
-    return {r: r, g: g, b: b};
+    return {r: r, g: g, b: b, a: a>0 ? a : 255};
```
and here:
```js
return {r: r, g: g, b: b, a: a > 0 ? a: 255};
```

thisandagain reviewed (changes requested):
Looks great! Two very small lint issues that are failing CI. You should be able to test this locally by running `npm test`.

griffpatch committed:
Fix lint issues

Had to update the test scripts to handle the alpha channel, also I note
that all the hex tests are using CSS notation, not scratch notation
(which is 0x not #)

griffpatch commented:
Ok, I've fixed the lint issues - I also had to add to the test scripts to handle the new alpha values. Do I need to do anything to move the pull request forward?

thisandagain commented:
Thanks @griffpatch !
